client/src/components/Editor.js

```diff
@@ -1,9 +1,62 @@
-import React, {useState} from 'react'
+import React, {useState, useRef, useEffect} from 'react'
+import { useDocument, useDocumentUpdate } from '../hooks/DocumentContext'
 import { Button } from './Button'
 import Carousel from './Carousel'
+import axios from 'axios'
 import styles from './Editor.module.css'
 
-export const Editor = ({ blocks, validateAll }) => {
+const END_BUTTON_CHILD = -1
+
+export const Editor = ({ blocks, validateAll, saveAll }) => {
+
+  const carouselRef = useRef(null)
+  const updateDocument = useDocumentUpdate()
+  const document = useDocument()
+  const [updated, setUpdated] = useState(false)
+
+  useEffect(() => {
+    if (saveAll) {
+      updateBlocks()
+      setUpdated(() => true)
+    }
+  }, [saveAll])
+
+  useEffect(() => {
+    if (updated) {
+      (document.untracked)
+      ? postDocument(document)
+      : putDocument(document)
+    }
+  }, [document])
```

Comment:
Yeah so those two useEffects are called every time saveAll and document changes however they are guarded by those two booleans. I have tested them and they don't cause infinite loops or anything since they are guarded by conditionals.
That said, I'm not very happy about them being there as I would prefer something more readable and scalable.
The reason why I opted for them is that the save/validate buttons are called in `Header.js` and pass up a function to `EditorPage.js`, until then no problem. The issue then is passing them down to the children and that is not possible with functions references, so I used states instead.

There are two solutions:
* Refactor it so that everything is done in the parent (`EditorPage.js`)
* Probably use some more elegant hooks or maybe even a Context to keep track of validate/save modes.